js/node_shared/unix.js

```diff
@@ -237,8 +237,13 @@ function unix_getcwd(vunit) {
 //Provides: unix_chdir
 //Requires: caml_sys_chdir
 function unix_chdir(path) {
+  /* When we do a Unix chdir, this signals that we would like for the Node
+     process to change directories too.
+     This ensures that after doing a `Unix.chdir`, calls to things like Git
+     will respond as if we were actually in those directories.
+   */
+  globalThis.process.chdir(path);
   caml_sys_chdir(path);
-
```

Comment:
What would be nice to comment is what is doing caml_sys_chdir? and what is the difference between
Unix.chdir and Sys.chdir?